Source/monster.h

```diff
@@ -308,12 +308,19 @@ struct Monster { // note: missing field _mAFNum
 	}
 
 	/**
-	 * @brief Returns the network identifier for this monster
+	 * @brief Returns the internal identifier for this monster
 	 *
 	 * This is currently the index into the Monsters array, but may change in the future.
 	 */
 	[[nodiscard]] size_t getId() const;
 
+	/**
+	 * @brief Returns the network identifier for this monster
+	 *
+	 * This is currently the index into the Monsters array, but may change in the future.
+	 */
+	[[nodiscard]] size_t getNetworkId() const;

```

Comment:
Wrong way round. `Monster::getId()` *is* the network/lookup ID for a monster instance. The use in leashed monsters is consistent with the use in network actions so the more generic name is suitable for that function.

For the use in StartDeathFromMonster and MonsterHitMonster the important concepts are "is this monster controlled by a player" and "who is the controlling player". This function addresses the second requirement.
```suggestion
	/**
	 * @brief Returns the identifier for the player who owns/controls this monster
	 *
	 * In Diablo 1 players can only control a single monster. The first four slots of the Monsters array are reserved for
	 * player golems so we currently reuse the monster network id here, but provide a separate function to allow for
	 * possible future changes.
	 */
	[[nodiscard]] size_t getOwningPlayerId() const;
```

Reply:
I don't think I really favor this since it would be easy to assume that the return value will be blank for monsters that aren't controlled.

Reply:
Yeah, I was basing this implementation on the missile source by reference PR with the same assumption that every use would be guarded. It can return an optional instead but at that point this function really needs to be addressed in its own PR.

Reply:
The change was reverted, I am resolving this one